In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
ls

 C 드라이브의 볼륨에는 이름이 없습니다.
 볼륨 일련 번호: E070-A7AC

 C:\Users\TFG5076XG\Documents\python\pandas 디렉터리

2021-11-26  오후 02:14    <DIR>          .
2021-11-26  오후 02:14    <DIR>          ..
2021-11-26  오후 02:14    <DIR>          .ipynb_checkpoints
2021-11-26  오후 01:53    <DIR>          output
2021-11-26  오후 01:58           437,385 pandas.ipynb
2021-11-26  오후 02:11            39,129 Pandas_퀴즈1.ipynb
2021-11-26  오후 02:14            66,960 Pandas_퀴즈2.ipynb
2021-11-24  오후 04:21         3,452,101 판다스.pdf
               4개 파일           3,995,575 바이트
               4개 디렉터리  440,367,603,712 바이트 남음


# Q1-1

```
아래의 파일을 이용하여 df_covid 데이타프레임을 생성하여라.
   
seoul-covid19_6_30_.csv    

```

In [2]:
df_covid = pd.read_csv('./data/seoul-covid19_6_30_.csv')
df_covid

연번     환자    확진일   거주지     여행력          접촉력         조치사항
0     1321  12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
1     1320  12829  6.30.   관악구       -         확인 중        보라매병원
2     1319  12831  6.30.  영등포구       -         확인 중      중앙대학교병원
3     1318  12808  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
4     1317  12809  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
...    ...    ...    ...   ...     ...          ...          ...
1316     5      9  1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
1317     4      7  1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
1318     3      6  1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
1319     2      5  1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
1320     1      2  1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)

[1321 rows x 7 columns]

# Q1-2
```
'연번' 컬럼은 삭제하고  '환자' 컬럼 값 기준으로 정렬한 후 '환자' 컬럼을 인덱스로 설정하여라. 
```

In [3]:
del df_covid['연번']
df_covid

환자    확진일   거주지     여행력          접촉력         조치사항
0     12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
1     12829  6.30.   관악구       -         확인 중        보라매병원
2     12831  6.30.  영등포구       -         확인 중      중앙대학교병원
3     12808  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
4     12809  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
...     ...    ...   ...     ...          ...          ...
1316      9  1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
1317      7  1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
1318      6  1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
1319      5  1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
1320      2  1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)

[1321 rows x 6 columns]

In [6]:
df_covid.sort_values?

Signature:
df_covid.sort_values(
    by: 'IndexLabel',
    *,
    axis: 'Axis' = 0,
    ascending: 'bool | list[bool] | tuple[bool, ...]' = True,
    inplace: 'bool' = False,
    kind: 'SortKind' = 'quicksort',
    na_position: 'str' = 'last',
    ignore_index: 'bool' = False,
    key: 'ValueKeyFunc | None' = None,
) -> 'DataFrame | None'
Docstring:
Sort by the values along either axis.

Parameters
----------
by : str or list of str
    Name or list of names to sort by.

    - if `axis` is 0 or `'index'` then `by` may contain index
      levels and/or column labels.
    - if `axis` is 1 or `'columns'` then `by` may contain column
      levels and/or index labels.
axis : "{0 or 'index', 1 or 'columns'}", default 0
     Axis to be sorted.
ascending : bool or list of bool, default True
     Sort ascending vs. descending. Specify list for multiple sort
     orders.  If this is a list of bools, must match the length of
     the by.
inplace : bool, default False
     If True, perform operati

In [10]:
df_covid = df_covid.sort_values(by='환자')
df_covid

확진일   거주지     여행력          접촉력         조치사항
환자                                                  
2      1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)
5      1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
6      1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
7      1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
9      1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
...      ...   ...     ...          ...          ...
12819  6.30.   마포구       -    대전 꿈꾸는 교회         서남병원
12822  6.29.   의정부       -         확인 중       타시도 이관
12829  6.30.   관악구       -         확인 중        보라매병원
12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
12831  6.30.  영등포구       -         확인 중      중앙대학교병원

[1321 rows x 5 columns]

# Q1-3
어느 지역에서 제일 많은 확진자가 발생했는지 1~5위 까지 출력하여라 (거주지 기준)

In [13]:
df_group = df_covid.groupby('거주지').count()
df_group
top5 = list(df_group["확진일"].nlargest(5, keep='all').index.values)
top5

['관악구', '구로구', '강남구', '강서구', '양천구']


# Q1-4
확진자가 가장 많이 발생한 1, 2, 3위 지역의 데이타만 출력하여라 

# Q1-5
'확진일' 컬럼값을 아래와 같은 형태로 변경하여라
<br>
'1.24.' => '2020-01-24'

# Q1-6
'확진일' 컬럼을 시계열 데이타로 형태를 변경하여라 

# Q1-7
3월에 발생한 확진자의 데이타만 출력하여라.

# Q1-8
주말에 발생한 확진자의 데이타만 출력하여라.

# Q1-9
'콜센터'와 관련된 확진자의 데이타를 출력하여라. 

## Q1-10
'우한'과 관련된 확진자의 데이타만 출력하여라. 

## Q1-11
사망자는 모두 몇명인가?

# Q2-1

아래의 파일을 이용하여 데이타프레임을 생성하여라.<br>
   
전국 평균 분양가격.csv 

# Q2-2

데이터프레임의 행과 열을 서로 변경하여라.

# Q2-3

1행을 삭제하고 2행을 컬럼으로 지정하여라. 

# Q2-4

'지역' 컬럼을 시계열 데이타로 변경하고 인덱스로 설정하여라. 

# Q2-5

2015년 6월 이후의 데이타만 출력하여라 

# Q2-6

2014년도의 서울, 인천 경기 컬럼만 출력하여라. 

# Q2-7
csv 파일 형태로 현재의 데이타프레임을 저장하여라. 
<br>
output/평균분양가격_서울인천경기.csv